apps/prairielearn/src/pages/publicAssessmentQuestionsPreview/publicAssessmentQuestionsPreview.ts

```diff
@@ -0,0 +1,143 @@
+
+import {
+  InstructorAssessmentQuestions,
+  AssessmentQuestionRowSchema,
+} from './publicAssessmentQuestionsPreview.html.js';
+
+import { z } from 'zod';
+
+async function selectAssessmentById(assessment_id: string): Promise<Assessment> {
+  return await queryRow(
+    sql.select_assessment_by_id,
+    {
+      assessment_id,
+    },
+    AssessmentSchema,
+  );
+}
+
+const BooleanSchema = z.boolean();
+
+async function checkAssessmentPublic(assessment_id: string): Promise<boolean> {
+  const isPublic = await queryRow(
+    sql.check_assessment_is_public,
+    { assessment_id },
+    BooleanSchema,
+  );
+  return isPublic;
+}
+
+async function checkCourseInstancePublic(course_instance_id: string): Promise<boolean> {
+  const isPublic = await queryRow(
+    sql.check_course_instance_is_public,
+    { course_instance_id },
+    BooleanSchema,
+  );
+  return isPublic;
+}
+
+const ansiUp = new AnsiUp();
+const router = express.Router();
+const sql = loadSqlEquiv(import.meta.url);
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const isCourseInstancePublic = await checkCourseInstancePublic(res.locals.course_instance_id);
+    const isAssessmentPublic = await checkAssessmentPublic(res.locals.assessment_id);
+    const courseId = await selectCourseIdByInstanceId(res.locals.course_instance_id.toString());
+    const course = await selectCourseById(courseId);
+    res.locals.course = course;
+
+    if (!isCourseInstancePublic && !isAssessmentPublic) {
+      throw new error.HttpStatusError(
+        404,
+        'The course instance that owns this assessment is not public.',
+      );
+    }
+
+    res.locals.assessment = await selectAssessmentById(res.locals.assessment_id);
+    const questionRows = await queryRows(
+      sql.questions,
+      {
+        assessment_id: res.locals.assessment.id,
+        course_id: res.locals.course.id,
+      },
+      AssessmentQuestionRowSchema,
+    );
+    const questions = questionRows.map((row) => {
+      if (row.sync_errors) row.sync_errors_ansified = ansiUp.ansi_to_html(row.sync_errors);
+      if (row.sync_warnings) row.sync_warnings_ansified = ansiUp.ansi_to_html(row.sync_warnings);
+      return row;
+    });
+    res.send(InstructorAssessmentQuestions({ resLocals: res.locals, questions }));
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (req.body.__action === 'copy_assessment') {
+      // Get needed data
+      const courseId = await selectCourseIdByInstanceId(res.locals.course_instance_id);
+      res.locals.course = await selectCourseById(courseId);
+      res.locals.course_instance = await selectCourseInstanceById(res.locals.course_instance_id);
+      res.locals.assessment = await selectAssessmentById(res.locals.assessment_id);
+
+      res.locals.assessment.sharedPublicly = true; // TEST
+      res.locals.user = { id: 'test' }; // TEST
+      res.locals.authz_data = { authn_user: { user_id: 'test' } }; // TEST
+
```

Comment:
clean up testing code here

Reply:
Do we want to remove it since we're not doing the copying functionality yet?

Reply:
I removed the copy button and its backend stuff